src/plugins/onesignal.ts

```diff
@@ -0,0 +1,254 @@
+import {Plugin, Cordova} from './plugin';
+
+/**
+ * @name OneSignal
+ * @description
+ * The OneSignal plugin is an client implementation for using the [OneSignal](https://onesignal.com/) Service.
+ * OneSignal is a simple implementation for delivering push notifications.
+ *
+ * Requires Cordova plugin: `onesignal-cordova-plugin`. For more info, please see the [OneSignal Cordova Docs](https://documentation.onesignal.com/docs/phonegap-sdk-installation).
+ *
+ * @usage
+ * ```ts
+ * import {OneSignal} from 'ionic-native';
+ *
+ * let notificationOpenedCallback(jsonData: any) {
+ *   console.log('didReceiveRemoteNotificationCallBack: ' + JSON.stringify(jsonData));
+ * };
+ * OneSignal.init("b2f7f966-d8cc-11e4-bed1-df8f05be55ba",
+ *                                {googleProjectNumber: "703322744261"},
+ *                                notificationOpenedCallback);
+ * 
+ * OneSignal.enableInAppAlertNotification(true);
+ * ```
+ *
+ */
+@Plugin({
+    plugin: 'onesignal-cordova-plugin',
+    pluginRef: 'plugins.OneSignal',
+    repo: 'https://github.com/OneSignal/OneSignal-Cordova-SDK',
+    platforms: ['Android', 'iOS', 'Windows Phone 8']
+})
+export class OneSignal {
+
+    /**
+     * Only required method you need to call to setup OneSignal to receive push notifications. Call this from the `deviceready` event.
+     *
+     * @param {appId} Your AppId from your OneSignal app
+     * @param {options} The Google Project Number (which you can get from the Google Developer Potal) and the autoRegister option.
+     * @param {object} This should be a callback function
+     * @returns {Promise} Returns a Promise that resolves when remote notification was recieved.
+     */
+    @Cordova()
+    static init(appId: string,
+        options: {
+            googleProjectNumber: string,
+            autoRegister: boolean
+        }, object?: any): void { }
+
+
+    /**
+     * Call this when you would like to prompt an iOS user to accept push notifications with the default system prompt. 
+     * Only use if you passed false to autoRegister when calling init.
+     */
+    @Cordova()
```

Comment:
Having `@Cordova()` decorator without any configuration assumes the function has `successCallback` and `errorCallback` as the last 2 parameters. This means that it will attempt to return a promise when this function is invoked.

This should be `@Cordova({ sync: true })` which means that it will call the function and return back the result as-is.